github/github.go

```diff
@@ -0,0 +1,41 @@
+package github
+
+import (
+	"context"
+
+	"release-manager/github/model"
+	"release-manager/github/util"
+	svcmodel "release-manager/service/model"
+
+	"github.com/google/go-github/v60/github"
+)
+
+type ClientFactory struct{}
+
+func NewClientFactory() *ClientFactory {
+	return &ClientFactory{}
+}
+
+type Client struct {
+	Client *github.Client
+}
+
+func (f *ClientFactory) NewClient(tkn string) svcmodel.GithubClient {
+	return &Client{
+		Client: github.NewClient(nil).WithAuthToken(tkn),
+	}
+}
+
+func (c *Client) ListTagsForRepository(ctx context.Context, repo svcmodel.GithubRepository, limit int) ([]svcmodel.GitTag, error) {
+	t, _, err := c.Client.Repositories.ListTags(
+		ctx,
+		repo.OwnerSlug,
+		repo.RepositorySlug,
+		&github.ListOptions{PerPage: limit},
```

Comment:
Don't we also want to specify the page?

Or use a much bigger limit and document it, for example right here, above the method.
(Obviously, would be considered a short-run solution, and "add pagination" would be a separate task, but that's fine.)

And since the `githubRepositoryTagsLimit` is used only in one place, maybe even simply move it here, to the client...

Reply:
In the end, I removed the list options configuration and kept the default settings. Now, 30 tags will be returned by default.

I think it's okay for the MVP. The last 30 tags should be more than enough since you'll usually choose the newest ones when linking the release to the source code. Additionally, if the client wants to display fewer tags, this can be handled on the client side.